nanoFramework.CoreLibrary/System/GC.cs

```diff
@@ -1,39 +1,79 @@
-        {
-            while (AnyPendingFinalizers()) Threading.Thread.Sleep(10);
-        }
+        /// <param name="forceFullCollection"><see langword="true"/> to indicate that this method can wait for garbage collection and heap compaction to occur before returning; otherwise, <see langword="false"/>.</param>
+        /// <returns>The heap size, in bytes, excluding fragmentation.</returns>
+        public static long GetTotalMemory(bool forceFullCollection) => Run(forceFullCollection);
 
         /// <summary>
-        /// Requests that the system not call the finalizer for the specified object.
+        /// Requests that the system call the finalizer for the specified object for which <see cref="SuppressFinalize"/> has previously been called.
         /// </summary>
-        /// <param name="obj">The object that a finalizer must not be called for. </param>
+        /// <param name="obj">The object that a finalizer must be called for.</param>
+        /// <exception cref="ArgumentNullException"><paramref name="obj"/> is <see langword="null"/>.</exception>
         [MethodImpl(MethodImplOptions.InternalCall)]
-        public static extern void SuppressFinalize(Object obj);
+        public static extern void ReRegisterForFinalize(object obj);
+
+        /// <summary>
+        /// Forces an immediate garbage collection of all generations.
+        /// </summary>
+        /// <remarks>
+        /// Use this method to try to reclaim all memory that is inaccessible. It performs a blocking garbage collection of all generations.
+        /// All objects, regardless of how long they have been in memory, are considered for collection; however, objects that are referenced in managed code are not collected. Use this method to force the system to try to reclaim the maximum amount of available memory.
+        /// </remarks>
+        public static void Collect() => Run(true);
 
         /// <summary>
-        /// Requests that the system call the finalizer for the specified object for which SuppressFinalize has previously been called.
+        /// Requests that the common language runtime not call the finalizer for the specified object.
         /// </summary>
-        /// <param name="obj">The object that a finalizer must be called for. </param>
+        /// <param name="obj">The object whose finalizer must not be executed.</param>
+        /// <exception cref="ArgumentNullException"><paramref name="obj"/> is <see langword="null"/>.</exception>
         [MethodImpl(MethodImplOptions.InternalCall)]
-        public static extern void ReRegisterForFinalize(Object obj);
+        public static extern void SuppressFinalize(object obj);
 
+        /// <summary>
+        /// Suspends the current thread until the thread that is processing the queue of finalizers has emptied that queue.
+        /// </summary>
+        public static void WaitForPendingFinalizers()
+        {
+            while (AnyPendingFinalizers()) Threading.Thread.Sleep(10);
```

Comment:
maybe just Thread.Spin() ? or will that too much blocking?

Reply:
Better to use Sleep, yes. As the thread running the finalizers needs time to complete.